src/platform/plugins/shared/workflows_execution_engine/server/step/kibana_action_step.ts

```diff
@@ -199,18 +211,135 @@ export class KibanaActionStepImpl extends BaseAtomicNodeImplementation<KibanaAct
+    if (fetcherOptions && Object.keys(fetcherOptions).length > 0) {
+      const { Agent } = await import('undici');
+
+      const {
+        // eslint-disable-next-line @typescript-eslint/naming-convention
+        skip_ssl_verification, // eslint-disable-next-line @typescript-eslint/naming-convention
+        follow_redirects, // eslint-disable-next-line @typescript-eslint/naming-convention
+        max_redirects, // eslint-disable-next-line @typescript-eslint/naming-convention
+        keep_alive,
+        timeout,
+        retry,
+        ...otherOptions
+      } = fetcherOptions;
 
-    if (!response.ok) {
-      const errorText = await response.text();
-      throw new Error(`HTTP ${response.status}: ${errorText}`);
+      const agentOptions: any = { ...otherOptions };
+
+      // Map our options to undici Agent options
+      if (skip_ssl_verification) {
+        agentOptions.connect = { ...(agentOptions.connect || {}), rejectUnauthorized: false };
+      }
+      if (max_redirects !== undefined) {
+        agentOptions.maxRedirections = max_redirects;
+      }
+      if (keep_alive !== undefined) {
+        agentOptions.keepAliveTimeout = keep_alive ? 60000 : 0;
+        agentOptions.keepAliveMaxTimeout = keep_alive ? 600000 : 0;
+      }
+
+      (fetchOptions as any).dispatcher = new Agent(agentOptions);
+
+      // Handle redirect at fetch level
+      if (follow_redirects === false) {
+        fetchOptions.redirect = 'manual';
+      }
+    }
+
+    // Setup timeout if configured
+    let timeoutId: NodeJS.Timeout | undefined;
+    if (fetcherOptions?.timeout) {
+      const controller = new AbortController();
+      timeoutId = setTimeout(() => controller.abort(), fetcherOptions.timeout);
+      fetchOptions.signal = controller.signal;
     }
 
-    const responseData = await response.json();
-    return responseData;
+    // Execute with retry logic
+    return this.executeWithRetry(fullUrl, fetchOptions, fetcherOptions?.retry, timeoutId);
+  }
+
+  private async executeWithRetry(
+    url: string,
+    fetchOptions: RequestInit,
+    retryConfig?: Record<string, any>,
+    timeoutId?: NodeJS.Timeout
+  ): Promise<any> {
+    const maxAttempts = retryConfig?.attempts ?? 1;
+    const retryDelay = retryConfig?.delay ?? 1000;
+    const backoffStrategy = retryConfig?.backoff ?? 'linear';
+    const retryOnStatuses = retryConfig?.retryOn ?? [502, 503, 504];
+
+    let lastError: any;
+    for (let attempt = 0; attempt < maxAttempts; attempt++) {
+      try {
+        const response = await fetch(url, fetchOptions);
+
+        if (timeoutId) clearTimeout(timeoutId);
+
+        // Retry on specific status codes
+        if (
+          !response.ok &&
+          attempt < maxAttempts - 1 &&
+          retryOnStatuses.includes(response.status)
+        ) {
+          const delay =
+            backoffStrategy === 'exponential' ? retryDelay * Math.pow(2, attempt) : retryDelay;
+          this.workflowLogger.logInfo(
```

Comment:
Wouldn't it be better to use the debug level?

Reply:
changed to debug